### PR TITLE
Migrate presentation list component

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Migrate presentation list component

Upgrade weekly-templates and batch-operations
Create presentationStatus pipe

Clean out extra functionality from controller
Controller is still used by two other lists

[stage-2]

## Motivation and Context
Angular migration epic

## How Has This Been Tested?
Tested locally.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Unfortunately this controller is used via `ng-controller` for the presentation selector modals, so it cannot be completely removed/migrated. We can discuss alternatives.

Also, the links to the Presentation Manage page are currently broken because of `ui-sref`, I will merge your release once available to enable those.

@ezequielc please review. Thanks!

File: src/app/editor/pipes/presentation-status.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'presentationStatus'
+})
+export class PresentationStatusPipe implements PipeTransform {
```
Comment (ezequielc) [inline src/app/editor/pipes/presentation-status.pipe.ts:6]: @alex-deaconu Are you planning to downgrade this and remove the ajs filter? Thanks

File: src/app/editor/presentation-list/presentation-list.component.html [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      </table>
+    </div>
+  </div>
+
+  <div class="border-container text-center u_margin-md-bottom" *ngIf="!presentations.loadingItems && presentations.items.list.length === 0 && search.query === undefined">
+    <div class="panel-body">
+      <div class="form-group my-4">
+        <label>Add a presentation to start communicating better with your audience!</label>
+      </div>
+      <button id="sharedTemplatesButton" (click)="editorFactory.addFromSharedTemplateModal()" class="btn btn-default btn-toolbar-wide mb-2">
```
Comment (ezequielc) [inline src/app/editor/presentation-list/presentation-list.component.html:91]: @alex-deaconu I noticed that there is no longer a spacing between this and  the next button. I see that classes are the same, so I am not sure what happened, but I think we need to add a margin here. Thanks
Sample: https://apps-stage-2.risevision.com/editor/list?cid=a49cc477-8de8-4369-94d9-f66bcb2b2736

**Review by ezequielc — APPROVED** [review]
@alex-deaconu Looks good, but some comments inline. Thanks

File: src/app/editor/pipes/presentation-status.pipe.ts (hunk 1 above)
Comment (alex-deaconu) [reply src/app/editor/pipes/presentation-status.pipe.ts:6]: @ezequielc I don't know if this filter makes sense to be honest, so I left it as is. Without the i18n string it just basically returns the original string or `N/A`. Thanks.

File: src/app/editor/presentation-list/presentation-list.component.html (hunk 2 above)
Comment (alex-deaconu) [reply src/app/editor/presentation-list/presentation-list.component.html:91]: @ezequielc interesting, I will fix. Thanks!

Comment (alex-deaconu) [reply src/app/editor/presentation-list/presentation-list.component.html]: @ezequielc this should be fixed. Thanks!

File: src/app/editor/pipes/presentation-status.pipe.ts (hunk 1 above)
Comment (ezequielc) [reply src/app/editor/pipes/presentation-status.pipe.ts:6]: @alex-deaconu Got it. Sounds good. Thanks

File: src/app/editor/presentation-list/presentation-list.component.html (hunk 2 above)
Comment (ezequielc) [reply src/app/editor/presentation-list/presentation-list.component.html]: @alex-deaconu Looks good. Thanks!

**Review by ezequielc — APPROVED** [review]

**Pull request merged**